pkg/runtime/reconciler.go

```diff
@@ -361,7 +362,7 @@ func (r *resourceReconciler) cleanup(
 	rlog.Enter("rm.Delete")
 	latest, err := rm.Delete(ctx, observed)
 	rlog.Exit("rm.Delete", err)
-	if latest != nil {
+	if latest != nil && !reflect.ValueOf(latest).IsNil() {
```

Comment:
I would recommend reusing this IsNil() method.
https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/compare/nil.go#L35

Reply:
Thanks @vijtrip2 
I have updated it to use ```IsNotNil()``` from: https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/compare/nil.go#L47